documentation for importing data with docker (#7530)
cBioPortal/cbioportal: docs/docker/import_data.md

## Import data instructions ##
This is an example to import a sample study: `study_es_0`.

> When try to import other studies, please follow the same routine:
> 1. import gene panels (if applicable, some studies may don't need to import gene panels)
> 2. import data

Reviewer: ```suggestion
When trying to import other studies, please follow the same routine:
1. import gene panels (if applicable, studies without gene panels are assumed to be whole exome/genome)
2. import study data
```

### Step 1 - Import gene panels

> To import gene panels for your study, please referece the example commands in [this file](example_commands.md#importing-gene-panel)

Reviewer: ```suggestion
To import gene panels for your study, please reference the example commands in [this file](example_commands.md#importing-gene-panel)
```

> Commands for importing `study_es_0` gene panels(`data_gene_panel_testpanel1` and `data_gene_panel_testpanel2`):

Reviewer: ```suggestion
These are the commands for importing `study_es_0` gene panels (`data_gene_panel_testpanel1` and `data_gene_panel_testpanel2`):
```

```shell
docker run -it --rm \
    --net cbio-net \
    -v /<path_to_config_file>/portal.properties:/cbioportal/portal.properties:ro \
    cbioportal/cbioportal:latest \
    bash -c 'cd /cbioportal/core/src/main/scripts/ && ./importGenePanel.pl --data /cbioportal/core/src/test/scripts/test_data/study_es_0/data_gene_panel_testpanel1.txt'
```

```shell
docker run -it --rm \
    --net cbio-net \
    -v /<path_to_config_file>/portal.properties:/cbioportal/portal.properties:ro \
    cbioportal/cbioportal:latest \
    bash -c 'cd /cbioportal/core/src/main/scripts/ && ./importGenePanel.pl --data /cbioportal/core/src/test/scripts/test_data/study_es_0/data_gene_panel_testpanel2.txt'
```

### Step 2 - Import data

> To import data for your study, please referece the example commands in [this file](example_commands.md#importing-data)

Reviewer: ```suggestion
To import data for your study, please reference the example commands in [this file](example_commands.md#importing-data)
```

Command for importing `study_es_0` data:

```shell
docker run -it --rm --net cbio-net \
    -v /<path_to_config_file>/portal.properties:/cbioportal/portal.properties:ro \
    cbioportal/cbioportal:latest \
    metaImport.py -u http://cbioportal-container:8080 -s /cbioportal/core/src/test/scripts/test_data/study_es_0
```

:warning: after importing a study, remember to restart `cbioportal-container`
to see the study on the home page. Run `docker restart cbioportal-container`.

> ## Common Questions and solutions

Reviewer: ```suggestion
## Frequently Asked Questions
```
I think this is the common title for these kind of sections

This is a great idea to include btw 👍

    cbioportal/cbioportal:latest \
    bash -c 'cd /cbioportal/core/src/main/scripts/ && ./importGenePanel.pl --data /cbioportal/core/src/test/scripts/test_data/study_es_0/data_gene_panel_testpanel1.txt'
```

```shell
docker run -it --rm \
    --net cbio-net \
    -v /<path_to_config_file>/portal.properties:/cbioportal/portal.properties:ro \
    cbioportal/cbioportal:latest \
    bash -c 'cd /cbioportal/core/src/main/scripts/ && ./importGenePanel.pl --data /cbioportal/core/src/test/scripts/test_data/study_es_0/data_gene_panel_testpanel2.txt'
```

### Step 2 - Import data

To import data for your study, please reference the example commands in [this file](example_commands.md#importing-data)

Command for importing `study_es_0` data:

```shell
docker run -it --rm --net cbio-net \
    -v /<path_to_config_file>/portal.properties:/cbioportal/portal.properties:ro \
    cbioportal/cbioportal:latest \
    metaImport.py -u http://cbioportal-container:8080 -s /cbioportal/core/src/test/scripts/test_data/study_es_0
```

:warning: after importing a study, remember to restart `cbioportal-container`
to see the study on the home page. Run `docker restart cbioportal-container`.

> 

Reviewer: Maybe good to have some concluding notes at the end. E.g.:

You have now imported the test study `study_es_0`. This study is included inside the cbioportal container. The process for adding a study that is outside of the container is almost the same. The difference is that one needs to add the `-v` parameter to mount the folder with the study data inside the container. See `How to import a real study` section (add markdown link) for an example.

## Frequently Asked Questions

### Gene panel ID is not in database

If you see an error like this when you importing the data:  
`ERROR: data_gene_panel_matrix.txt: lines [2, 3, 4, (10 more)]: Gene panel ID is not in database. Please import this gene panel before loading study data.; values encountered: ['TESTPANEL1', 'TESTPANEL2']`  

please follow the first step to import gene panels (e.g. import `data_gene_panel_testpanel1` and `data_gene_panel_testpanel2` for `study_es_0`), then try to import the data again.

### Study imported correctly, but got error when trying to query something

Remember to restart the `cbioportal-container` after data imported.
```shell
docker restart cbioportal-container
```

### How to import a real study

> You can find all public studies at [cbioportal.org](http://www.cbioportal.org/data_sets.jsp) a zipped folder with staging files from each study can be downloaded. These zip files are compressed versions of the study folders in [datahub](https://github.com/cBioPortal/datahub).
> 
> Take study `Cholangiocarcinoma (TCGA, PanCancer Atlas)` as an example:  
> Study `Cholangiocarcinoma (TCGA, PanCancer Atlas)` is a whole exome/genome study, so just need to download the study data and run the command from step2.

Reviewer: ```suggestion
You can find all public studies at [cbioportal.org](http://www.cbioportal.org/data_sets.jsp) a zipped folder with staging files from each study can be downloaded. These zip files are compressed versions of the study folders in [datahub](https://github.com/cBioPortal/datahub).

Download and extract the zip file for study `Cholangiocarcinoma (TCGA, PanCancer Atlas)`. Study `Cholangiocarcinoma (TCGA, PanCancer Atlas)` is a whole exome study. You can therefore skip step 1 from the import data instructions section (add markdown link): the default gene panel is whole exome. so there is no need to import a custom one. Proceed with the command from step2
```
